etc/elasticsearch.yml

```diff
@@ -7,97 +7,82 @@
 #
-# Please see the documentation for further information on configuration options:
-# <http://www.elastic.co/guide/en/elasticsearch/reference/current/setup-configuration.html>
+# Please consult the documentation for further information on configuration options:
+# https://www.elastic.co/guide/en/elasticsearch/reference/index.html
 #
 # ---------------------------------- Cluster -----------------------------------
-
+#
 # Use a descriptive name for your cluster:
-
-cluster.name: ${CLUSTER_NAME}
-
+#
+cluster.name: ${ES_CLUSTER_NAME}
+#
 # ------------------------------------ Node ------------------------------------
-
-# Assign the container ID to the node name so to eliminate having to translate
-# between the two while debugging.
-
-node.name: es-${HOSTNAME}
-
-# By default, all nodes are eligible to become masters and store data.
-
-node.master: ${ES_NODE_MASTER}
-node.data: ${ES_NODE_DATA}
-
+#
+# Use a descriptive name for the node:
+#
+node.name: node-${HOSTNAME}
+#
 # Add custom attributes to the node:
 #
-# node.rack: r1
-
+#node.attr.rack: r1
+#
 # ----------------------------------- Paths ------------------------------------
-
+#
 # Path to directory where to store the data (separate multiple locations by comma):
-
-path.data: /var/lib/elasticsearch/data
-
+#
+#path.data: /path/to/data
+#
 # Path to log files:
-
-path.logs: /var/log/elasticsearch
-
+#
+#path.logs: /path/to/logs
+#
 # ----------------------------------- Memory -----------------------------------
 #
 # Lock the memory on startup:
 #
-# bootstrap.mlockall: true
+#bootstrap.memory_lock: true
 #
-# Make sure that the `ES_HEAP_SIZE` environment variable is set to about half the memory
-# available on the system and that the owner of the process is allowed to use this limit.
+# Make sure that the heap size is set to about half the memory available
+# on the system and that the owner of the process is allowed to use this
+# limit.
 #
 # Elasticsearch performs poorly when the system is swapping the memory.
 #
 # ---------------------------------- Network -----------------------------------
-
+#
 # Set the bind address to a specific IP (IPv4 or IPv6):
-
+#
 network.host: _eth0:ipv4_
-
+#
 # Set a custom port for HTTP:
 #
-# http.port: 9200
+#http.port: 9200
 #
-# For more information, see the documentation at:
-# <http://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html>
+# For more information, consult the network module documentation.
 #
-# ---------------------------------- Gateway -----------------------------------
+# --------------------------------- Discovery ----------------------------------
 #
-# Block initial recovery after a full cluster restart until N nodes are started:
+# Pass an initial list of hosts to perform discovery when new node is started:
+# The default list of hosts is ["127.0.0.1", "[::1]"]
 #
-# gateway.recover_after_nodes: 3
+discovery.zen.ping.unicast.hosts: ["${ES_BOOTSTRAP_HOST}"]
```

Comment:
So I'll comment out that line and the function `replace` (which has been renamed `replaceZenHosts` to be a little more verbose) will place the host and uncomment it.

Something I just realized is that we only populate it with a single host. If that container restarts and the referenced node is no longer part of the cluster it will fail to join the cluster.